pkg/controller/namespace/namespace_controller.go

```diff
@@ -263,67 +234,73 @@ func (r *ReconcileNamespace) initRoles(namespace *corev1.Namespace) error {
 		// scope.kubesphere.io/namespace: ""
 		labelKey = fmt.Sprintf(iamv1alpha2.ScopeLabelFormat, iamv1alpha2.ScopeNamespace)
 	}
-	err := r.List(context.Background(), &roleBases, client.MatchingLabelsSelector{Selector: labels.SelectorFromSet(labels.Set{labelKey: ""})})
-	if err != nil {
-		klog.Error(err)
+	if err := r.List(ctx, &roleBases, client.MatchingLabelsSelector{Selector: labels.SelectorFromSet(labels.Set{labelKey: ""})}); err != nil {
+		logger.Error(err, "list role bases failed")
 		return err
 	}
-
 	for _, roleBase := range roleBases.Items {
-		var role rbacv1.Role
-		if err = yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer(roleBase.Role.Raw), 1024).Decode(&role); err == nil && role.Kind == iamv1alpha2.ResourceKindRole {
+		var template rbacv1.Role
+		if err := yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer(roleBase.Role.Raw), 1024).Decode(&template); err == nil && template.Kind == iamv1alpha2.ResourceKindRole {
 			var old rbacv1.Role
-			err := r.Client.Get(context.Background(), types.NamespacedName{Namespace: namespace.Name, Name: role.Name}, &old)
-			if err != nil {
+			if err := r.Client.Get(ctx, types.NamespacedName{Namespace: namespace.Name, Name: template.Name}, &old); err != nil {
 				if errors.IsNotFound(err) {
-					role.Namespace = namespace.Name
-					err = r.Client.Create(context.Background(), &role)
-					if err != nil {
-						klog.Error(err)
+					template.Namespace = namespace.Name
+					logger.V(4).Info("init builtin role", "role", template.Name)
+					if err := r.Client.Create(ctx, &template); err != nil {
+						logger.Error(err, "create role failed")
 						return err
 					}
 					continue
 				}
 			}
+			if !reflect.DeepEqual(template.Labels, old.Labels) ||
+				!reflect.DeepEqual(template.Annotations, old.Annotations) ||
+				!reflect.DeepEqual(template.Rules, old.Rules) {
 
-			if !reflect.DeepEqual(role.Labels, old.Labels) ||
-				!reflect.DeepEqual(role.Annotations, old.Annotations) ||
-				!reflect.DeepEqual(role.Rules, old.Rules) {
+				old.Labels = template.Labels
+				old.Annotations = template.Annotations
+				old.Rules = template.Rules
 
-				old.Labels = role.Labels
-				old.Annotations = role.Annotations
-				old.Rules = role.Rules
-
-				if err := r.Update(context.Background(), &old); err != nil {
-					klog.Error(err)
+				logger.V(4).Info("update builtin role", "role", template.Name)
+				if err := r.Update(ctx, &old); err != nil {
+					logger.Error(err, "update role failed")
 					return err
 				}
 			}
+		} else if err != nil {
+			logger.V(4).Info("invalid role base found", "name", roleBase.Name, "err", err.Error())
```

Comment:
The Log level should be increased. Please ref to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md

Reply:
done